<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.313">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Professor Peter Kille">
<meta name="dcterms.date" content="2023-02-19">

<title>Re-analysis of Published Data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<style>
    .quarto-title-block .quarto-title-banner {
      background: darkred;
    }
    </style>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Re-analysis of Published Data</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title d-none d-lg-block">Re-analysis of Published Data</h1>
            <p class="subtitle lead">NCBI Geo / Geo2R</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Professor Peter Kille </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">February 19, 2023</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation docked overflow-auto">
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">Omics an Introduction</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">The Omics Revolution</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Lung_model.html" class="sidebar-item-text sidebar-link">Introduction to the Lung</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">Viral Genomics</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Viral_Genomics.html" class="sidebar-item-text sidebar-link">Viral Phylogenomics</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">Microbiome Analysis</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./the_CF_lung.html" class="sidebar-item-text sidebar-link">The Microbiome of the CF Lung</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Microbiome.html" class="sidebar-item-text sidebar-link">Microbiome Data in QIIME</a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">Transcriptomics</a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Lungs_and_particles.html" class="sidebar-item-text sidebar-link">Lungs and Particles</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Transcriptomics_an_introduction.html" class="sidebar-item-text sidebar-link">Transcriptomics</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./NCBI_GEO.html" class="sidebar-item-text sidebar-link active">Re-analysis of Published Data</a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Functional_Interpretation.html" class="sidebar-item-text sidebar-link">Functional Interpretation of transcritomic data</a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#workshop-overview" id="toc-workshop-overview" class="nav-link active" data-scroll-target="#workshop-overview">Workshop Overview</a>
  <ul class="collapse">
  <li><a href="#processing-rnaseq-data" id="toc-processing-rnaseq-data" class="nav-link" data-scroll-target="#processing-rnaseq-data">Processing RNAseq Data</a></li>
  <li><a href="#dataset-to-use---geogds2565" id="toc-dataset-to-use---geogds2565" class="nav-link" data-scroll-target="#dataset-to-use---geogds2565">Dataset to use - Geo:GDS2565</a></li>
  <li><a href="#panopto_walk_through" id="toc-panopto_walk_through" class="nav-link" data-scroll-target="#panopto_walk_through">Panopto_walk_through</a></li>
  </ul></li>
  <li><a href="#key-statistical-concepts" id="toc-key-statistical-concepts" class="nav-link" data-scroll-target="#key-statistical-concepts">Key Statistical Concepts</a>
  <ul class="collapse">
  <li><a href="#false-discover-rate" id="toc-false-discover-rate" class="nav-link" data-scroll-target="#false-discover-rate">False Discover Rate</a></li>
  <li><a href="#fishers-exact-test-and-enrichment-analysis" id="toc-fishers-exact-test-and-enrichment-analysis" class="nav-link" data-scroll-target="#fishers-exact-test-and-enrichment-analysis">Fisher’s Exact Test and Enrichment Analysis</a></li>
  </ul></li>
  <li><a href="#mining-geo-database" id="toc-mining-geo-database" class="nav-link" data-scroll-target="#mining-geo-database">Mining GEO database</a>
  <ul class="collapse">
  <li><a href="#geo-workshop" id="toc-geo-workshop" class="nav-link" data-scroll-target="#geo-workshop">Geo Workshop</a></li>
  <li><a href="#geo-database-panopto" id="toc-geo-database-panopto" class="nav-link" data-scroll-target="#geo-database-panopto">GEO Database Panopto</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<section id="workshop-overview" class="level1">
<h1>Workshop Overview</h1>
<section id="processing-rnaseq-data" class="level2">
<h2 class="anchored" data-anchor-id="processing-rnaseq-data">Processing RNAseq Data</h2>
<p>Converting RNASeq data into gene counts is the first-step prior to analysis of the biological function and networks revealed through the subsequent transcription analysis. This process is outside the scope of this workshop but if you are interested <a href="https://cf.sharepoint.com/:b:/t/SystemsBiology/EQ_8U8YJ9i9Fg0bKPo6PmO4Bec8KxoXp6zIco-L5z-_55g?e=Hme6hj0">Andres et al 2013</a> Provides an excellent overview of the processes involved.</p>
<p><img src="images/Process_Overview.png" class="img-fluid" alt="Process Overview"> Figure 1. Process overview</p>
</section>
<section id="dataset-to-use---geogds2565" class="level2">
<h2 class="anchored" data-anchor-id="dataset-to-use---geogds2565"><a href="https://www.ncbi.nlm.nih.gov/sites/GDSbrowser?acc=GDS2565">Dataset to use - Geo:GDS2565</a></h2>
</section>
<section id="panopto_walk_through" class="level2">
<h2 class="anchored" data-anchor-id="panopto_walk_through"><a href="https://cardiff.cloud.panopto.eu/Panopto/Pages/Viewer.aspx?id=9d81b942-6a9a-42e5-8fd8-ac7401144342">Panopto_walk_through</a></h2>
</section>
</section>
<section id="key-statistical-concepts" class="level1">
<h1>Key Statistical Concepts</h1>
<p>The workshop will use tools that exploit statistical approaches to identify differential expressed genes (DEGs) and subsequently identify the biological pathways or processes that are over represented (some time refereed to as enriched) within the DEGs. You will not need to derive the mathematical formula underpinning these concepts or deploy the algorithums from first principles since the software you will use does this for you but it is worth knowing the background. There is an excellent academic from Stanford University Name Prof Josh Starmer who runs a youtube channel called <a href="https://www.youtube.com/channel/UCtYLUTtgS3k1Fg4y5tAhLbw">StatQuest</a> this deal with a large range of biological stats and has some great annotations to explain them, I am going to suggest you watch two of his videos to orientate you about the key statistical concepts for this session:</p>
<section id="false-discover-rate" class="level2">
<h2 class="anchored" data-anchor-id="false-discover-rate">False Discover Rate</h2>
<iframe data-external="1" src="https://www.youtube.com/watch?v=K8LQSvtjcEo&amp;t=404svem">
</iframe>
</section>
<section id="fishers-exact-test-and-enrichment-analysis" class="level2">
<h2 class="anchored" data-anchor-id="fishers-exact-test-and-enrichment-analysis">Fisher’s Exact Test and Enrichment Analysis</h2>
<iframe data-external="1" src="https://www.youtube.com/watch?v=udyAvvaMjfM">
</iframe>
</section>
</section>
<section id="mining-geo-database" class="level1">
<h1>Mining GEO database</h1>
<section id="geo-workshop" class="level2">
<h2 class="anchored" data-anchor-id="geo-workshop"><a href="https://cf.sharepoint.com/:b:/t/SystemsBiology/ESRyTPUlb_dJgpHLXUbYk2wBO_SyL9gXhOXOs-BpHFSPsw?e=FXrOLC">Geo Workshop</a></h2>
<ol type="1">
<li>Search for you dataset</li>
</ol>
<p>Go to Geo DataSets: (https://www.ncbi.nlm.nih.gov/gds/)[https://www.ncbi.nlm.nih.gov/gds/]</p>
<p><img src="images/GEO_DataSets.png" class="img-fluid" alt="GEO_DataSet Search"> <strong><em>GEO_DataSet Search</em></strong></p>
<p>now select <search></search></p>
<ol start="2" type="1">
<li>Refine search to only show ‘DataSets’ and ‘Series’</li>
</ol>
<p>Select <strong><em>&lt; DataSets and Series &gt;</em></strong> from the left hand menu.</p>
<p><img src="images/DataSet_Selection.png" class="img-fluid" alt="Refine to DataSets"> <strong><em>Refine to DataSets</em></strong></p>
<p>3A. Select procesed dataset (should be number 3 in list and have a heat map icon to the right)</p>
<p>Click on title <strong><em>&lt; Endothelial cell response to ultrafine particles &gt;</em></strong> to select DataSet</p>
<p><img src="images/GDS2565.png" class="img-fluid" alt="Entry Page"> <strong><em>Entry Page</em></strong></p>
<p>Take notes off all pertinent information about the experiment, including species and what microarray platform was used. In this case experiment was conducted on rats and analysed on an Affymetrix Human Genome U133 Plus 2.0 Array. Download any publication available. Also, if you are interested you could have a look at the cluster analysis on the right hand site. This will show you how the relationship of the expression profiles from each sample relative to each other. If the experiment was successful, all samples of a certain treatment should cluster together.</p>
<p>4A. Compare experiment samples</p>
<p>Click <strong><em>&lt;Compare 2 sets of sample&gt;</em></strong></p>
<p>Choose <strong><em>&lt;test e.g.&nbsp;One-tailed t-test (A &gt; B)&gt;</em></strong></p>
<p>Choose <significance level="">_ e.g.&nbsp;0.001</significance></p>
<p>Click on: Step 2: Select which Samples to put in Group A and Group B</p>
<p><img src="images/group_compare.png" class="img-fluid" alt="Select Groups to Comnpare"> <strong><em>Select Groups to compare</em></strong></p>
<p>Choose <strong><em>&lt; Query Group A vs B &gt;</em></strong></p>
<p>You should now see a list of the following DEGS</p>
<p><img src="images/DEGs.png" class="img-fluid" alt="DEG list"> <strong><em>DEG List</em></strong></p>
<p>5A. Download DEGs</p>
<p>You have &gt;100 DEGs but the page only displays the first 20. Before downloading DEGS change the items per page from 20 to 500.</p>
<p><img src="images/Items_per_page.png" class="img-fluid" alt="Items Per Page"> <strong><em>Items Per Page</em></strong></p>
<p>Select <strong><em>&lt; Download profile data &gt;</em></strong></p>
<p>and you will be prompted to save the profiling of the DEGs displayed as default file name <strong><em>&lt;profile_data.txt&gt;</em></strong></p>
<p>Save to appropriate location. If needed Select Page 2…. of the DEGS and repeat the download process.</p>
<p>Repeat this process for <strong><em>&lt;test e.g.&nbsp;One-tailed t-test (B &gt; A) significance level 0.001 &gt;</em></strong>_</p>
<p>6A. Open and Merge DEG lists in Excel</p>
<p>The DEG lists show the gene list with there relative expression level (normalised) and annotation for the genes involved (annotation shown in columns BG -&gt;). For Our next steps we will use the Gene symbol that is in Column BH.</p>
<p>3B. Select un-processed series (should be number 2 - it will have icon <strong><em>&lt; Analyze with GEO2R &gt;</em></strong> at the end of the entry)</p>
<p>Click on <strong><em>&lt; Analyze with GEO2R &gt;</em></strong></p>
<p>Now Define groups by click clicking the <strong><em>&lt; Define Group pull &gt;</em></strong> down and create groups ‘control’ and ‘Treatment’ (enter group name and press enter). Click on each sample in list and associated it with one of your group (you can hold ctrl down to select multiple entry before associating them with the group).</p>
<p><img src="images/compare_groups_geo2R.png" class="img-fluid" alt="Select Groups to Compare"> <strong><em>Select Groups to compare</em></strong></p>
<p>4B. Customise the Option and Analyse</p>
<p>Select <strong><em>&lt; Options &gt;</em></strong> and customise as shown below:</p>
<p><img src="images/Geo2R_Options.png" class="img-fluid" alt="Geo2R Options"> <strong><em>Geo2R Options</em></strong></p>
<p>Select <strong><em>&lt; reanalyze &gt;</em></strong></p>
<p>Will will now see a Processing icon - this may take a minute or two.</p>
<p>5B. DEGs</p>
<p>You will now see a table and a series of Visualisations - review the visualisation taking note of what each are showing you.</p>
<p><img src="images/GEO2R_results.png" class="img-fluid" alt="Geo2R results"> <strong><em>Geo2R results</em></strong></p>
<p>Venn diagram showing GSE4567: Limma, Padj&lt;0.05 - 704 genes - this is the set we will use</p>
<p>Click on <strong><em>&lt; Explore and download, control vs treatment and Download Significant genes &gt;</em></strong></p>
<p>This will give you a Tsv you can open in excel</p>
<p>6B DEG TSV</p>
<p>Open the DEG TSV - you will have ID (Affymetrix), Gene Symbol, Description and Log2(fold Change) and Adjusted p-value. The Gene.symbol can have multiple symonyms for each gene - this can bias future analsyis. Copy/paste gene symbol column to rught hand column (so no other data is on right), and use <strong><em>&lt; Data &gt; Text to Columns &gt; Delimited &gt; Other &gt; ‘/’ &gt; Finish &gt;</em></strong>_ to push secondary symbols into other columns.</p>
</section>
<section id="geo-database-panopto" class="level2">
<h2 class="anchored" data-anchor-id="geo-database-panopto"><a href="https://cardiff.cloud.panopto.eu/Panopto/Pages/Viewer.aspx?id=fc9e75dd-2cad-4c45-9c90-ab6b00bf41e2">GEO Database Panopto</a></h2>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>